financial_calculations/bond_cash_flows.py

```diff
@@ -45,9 +46,7 @@ def create_semi_bond_cash_flows(effective_date, balance, coupon, maturity_years)
         effective_date = datetime.strptime(effective_date, "%Y-%m-%d")
 
     # Convert effective date to a datetime object if it's a datetime64[D] object
-    if isinstance(effective_date, np.datetime64):
-        effective_date = effective_date.astype('datetime64[D]').tolist()
-        print(effective_date)
+    effective_date = convert_to_datetime(effective_date)
```

Comment:
this is what I was talking about last night.  I think the array version of this is named *_to_datetime64